### DISCUSSION TITLE
Are there plans for a Reactive implementation

### DISCUSSION THREAD

**Discussion by mjohnston-il** [graphql-java-generator/graphql-maven-plugin-project#152] [Ideas]
Hi all,
We're currently using the latest version in Micronaut but the generated code signatures requires us to block on the remote call.  Have you considered adding a reactive programming approach that returns Mono/Flux objects?

**Comment by etienne-sf** [comment]
Hello,

Hum, not yet.  

I'm working on the 2.x version, to use the spring-graphql dependency. 

Once this is done, I guess that it would be a good target. 

But I must first check what would be the impact (on both the plug-in itself, and the code of the application that would use the code)

Étienne

**Reply by teragrid** [reply]
Hi team, 

Is there any update yet on the Reactive implementation?

Thanks,

**Comment by etienne-sf** [comment]
Hello,

The 2.x version is based on webflux, so it should use the reactive implementation.

But, when checking before answering your question, I discovered this: if you're using subscription, then you have to add the `spring-boot-starter-websocket`, which (by default) add the `spring-webmvc` dependency.


And there is an issue in the doc about the mvc being used.
I'll check all this and get back to you.

**Reply by etienne-sf** [reply]
Hello, after reading the code once more, my previous answer is partially wrong. 

The code uses the ´web flux´ dependency. 
But the code that uses the generated code doesn't have access to the GraphQL reactive stuff. 
So your code won't be reactive. 

I'll change that. But it may be a complex change.

**Comment by etienne-sf** [comment]
The Reactive implementation is now commited. It will be available in the next release, coming soon.

The implementation now generates:
* Two executors for each query, mutation and subscription: the old one, non reactive, and a reactive one.
* The GraphQLRepositories are also available in reactive mode.

Etienne

**Reply by teragrid** [reply]
Thank you so much for your update! I will check on my side to see if any issues
